### PR TITLE
Upgrade to Angular 16

### PR DESCRIPTION
**Pull request by coverbeck**
**Description**
Upgrades to Angular 16.

**Review Instructions**
Make sure the site is working correctly. Ideally the smoke tests should catch anything.

**Issue**
SEAB-6140

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -19,6 +19,9 @@
         "plugin:@angular-eslint/recommended",
         "plugin:@angular-eslint/template/process-inline-templates"
       ],
+      "plugins": [
```
Comment (coverbeck) [inline .eslintrc.json:22]: I'm a little unclear on why this is now needed or why it wasn't needed before. `npm run lint` fails now without it.

File: src/app/mytools/my-tool/my-tool.component.html [hunk 2]
```diff
@@ -143,7 +143,7 @@
                     class="accent-1-dark-btn"
                     type="button"
                     (click)="addToExistingTools()"
-                    [disabled]="(isRefreshing$ | async) || (userQuery.user$ | async) === false"
+                    [disabled]="(isRefreshing$ | async) || (userQuery.user$ | async) === null"
```
Comment (coverbeck) [inline src/app/mytools/my-tool/my-tool.component.html:146]: The compiler started complaining about this, rightly so, `userQuery.user$`'s type is not a boolean. There are other similar instances of this that were also changed.

File: src/app/mytools/my-tool/my-tool.component.ts [hunk 3]
```diff
@@ -150,6 +150,7 @@ export class MyToolComponent extends MyEntry implements OnInit {
     this.router.events
       .pipe(
         filter((event) => event instanceof NavigationEnd),
+        map((event) => event as NavigationEnd),
```
Comment (coverbeck) [inline src/app/mytools/my-tool/my-tool.component.ts:153]: Compiler complained without adding this. I guess Angular 16 compiler is stricter about some things.

File: package.json [hunk 4]
```diff
@@ -116,11 +116,14 @@
...
     "prettier": "^2.3.1",
     "ts-node": "^10.9.1",
-    "typescript": "^4.8.4"
+    "typescript": "^4.9.5"
   },
   "overrides": {
     "@schematics/update": {
       "ini": "^1.3.8"
+    },
+    "ngx-mat-select-search": {
```
Comment (coverbeck) [inline package.json:125]: There is a newer ngx-mat-select-search that works out of the box with Angular 16, but it uses the new Material components, and we're still on the legacy material components.

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**